WeScan/Scan/CaptureSessionManager.swift

```diff
@@ -122,6 +128,18 @@ final class CaptureSessionManager: NSObject, AVCaptureVideoDataOutputSampleBuffe
     }
     
     internal func capturePhoto() {
+        
+        let captureConnection = photoOutput.connections.first { (connection) -> Bool in
+            return connection.inputPorts.first(where: { (port) -> Bool in
+                port.mediaType == .video
+            }) != nil
+        }
+        guard captureConnection != nil || captureConnection?.isEnabled == true || captureConnection?.isActive == true else {
```

Comment:
Shouldn't we really check for *all* of these to be true to be safe?

Reply:
Can we also remove `== true` to make the code cleaner?

Reply:
1> I think we'd better check all because Apple gives this type an array.

2> `captureConnection` is an optional value, so if remove `== true` we must unwrapped `captureConnection`. 

I think here we should not use  `guard`, use `if` more in line with our thinking. @Boris-Em 

```
 if captureConnection == nil || captureConnection?.isEnabled == flase || captureConnection?.isActive == false else {
            let error = ImageScannerControllerError.capture
            delegate?.captureSessionManager(self, didFailWithError: error)
            return
        }
```

@justJS thanks for you comment.

Reply:
If you change the first `captureConnection != nil` to `guard let captureConnection = captureConnection` you can continue using the unwrapped variant, remove the `== true` and check for both `isEnabled` and `isActive`:

```swift
        guard let captureConnection = captureConnection, captureConnection.isEnabled, captureConnection.isActive else {
```